build/log_shutdown.go

```diff
@@ -0,0 +1,59 @@
+package build
+
+import (
+	"github.com/btcsuite/btclog"
+	"github.com/lightningnetwork/lnd/signal"
+)
+
+// ShutdownLogger wraps an existing logger with a shutdown function which will
+// be called on Critical/Criticalf to prompt shutdown.
+type ShutdownLogger struct {
+	btclog.Logger
+
+	// shutdown is the shutdown function we will use to signal shutdown.
+	shutdown func()
+}
+
+// NewShutdownLogger creates a shutdown logger for the log provided which will
+// use the signal package to request shutdown on critical errors.
+func NewShutdownLogger(logger btclog.Logger) *ShutdownLogger {
+	shutdown := func() {
+		// If our application is not listening for interrupts, we return
```

Comment:
Can't this just be a regular method body given it closes over the logger which is already embedded as an attribute in the struct?